docs/manual/constant-value-checker.tex

```diff
@@ -334,6 +334,27 @@
 that possibility, or verify the lack of overflow using some other tool or
 manual analysis.
 
+\sectionAndLabel{Strings can be null in concatenations}{null-strings-concats}
+
+The Constant Value Checker considers all elements of a String concatenation
+non-null by default. The \code{-AnullStringsConcatenation} command-line option
+makes the checker more sound by analyzing the possibility of each element to
+be null. Only String literals, primitive types and effectively final variables
+are considered non-null. Figure~\ref{fig-null-strings-concat} gives an example
```

Comment:
Why are effectively final variables considered non-null?  Couldn't type be null, too?

Reply:
In a String concatenation, operands that are effectively final variables and are not parameters are considered non-null. Can you please give one example in which the contrary holds?

If I got it wrong, then my initial intention was to consider constant operands of String concatenation non-null, like in `"hello" + Integer.MAX_VALUE`. Then I extended it to every effectively final variable (initialized and unchanged).

Reply:
final String s = null;

Reply:
The fact that a variable is final (or effectively-final) has nothing to with whether or not a variable is null.

Reply:
See test case in 56daab506.

Reply:
Added  commit in which I changed the `isNullable` method to reflect what I said earlier here.

Reply:
Ah! compile-time constant are non-null.  Yes, I agree with that.  Please update the manual here to say that.